request_handlers.cc

```diff
@@ -0,0 +1,37 @@
+#include "request_handlers.h"
+
+#include <string>
+#include <iostream>
+
+using namespace lightning_server;
+
+const int MAX_REQ_SIZE = 8192;
+
+// Read in the request from the socket used to construct the handler, then
+// fill the response buffer with the necessary headers followed by the
+// original request.
+size_t request_handlers::echoRequestHandler(boost::shared_ptr<bai::tcp::socket> socket) {
+
+  // Read in request
+  char request_buffer[MAX_REQ_SIZE];
+  boost::system::error_code ec;
+  std::size_t bytes_read = socket->read_some(boost::asio::buffer(request_buffer), ec);
+  std::cout << "~~~~~~~~~~Request~~~~~~~~~~\n" << request_buffer << std::endl;
```

Comment:
I like that the debug output headers you use here follow SimpleHTTPServer's output. It means less of a mental context-switch when I'm comparing against that reference implementation. 😄